shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -353,13 +342,14 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
      gpuCB: ColumnarBatch): List[ParquetCachedBatch] = {
     // NOTE: this doesn't take nulls into account so we could be over-estimating the actual size
     // but that's still better than under-estimating
-    val estimatedRowSize = scala.Range(0, gpuCB.numCols()).map { index =>
-      gpuCB.column(index).dataType().defaultSize
+    val estimatedRowSize = scala.Range(0, gpuCB.numCols()).map { idx =>
+      gpuCB.column(idx).asInstanceOf[GpuColumnVector].getBase.getDeviceMemorySize / gpuCB.numRows()
     }.sum
     val rowsInBatch = (BYTES_ALLOWED_PER_BATCH / estimatedRowSize).toInt
     val splitIndices = scala.Range(rowsInBatch, gpuCB.numRows(), rowsInBatch)
-    val cudfTables = if (splitIndices.nonEmpty) {
-      val splutVectors = scala.Range(0, gpuCB.numCols()).map { index =>
+    val buffers = new ListBuffer[ParquetCachedBatch]
```

Comment:
buffers needs a `closeOnExcept` or equivalent to not leak on excption.

Reply:
hmm... not sure if I am missing something here. The batches have data on the Heap, so in case of an Exception, won't they be reclaimed by jvm?

Reply:
Sorry I got confused by what needed to be closed and what did not.